src/strategies/spirit/index.ts

```diff
@@ -0,0 +1,99 @@
+import { multicall } from '../../utils';
+import { formatUnits } from '@ethersproject/units';
+
+export const author = 'chimpydev';
+export const version = '0.1.0';
+
+async function search_contract_cretion_block(contract_address, provider) {
+  var highest_block = await provider.getBlockNumber();
+  var lowest_block = 13691033;
+
+  var contract_code = await provider.getCode(contract_address, highest_block);
+
+  if (contract_code == '0x') {
+    return -1;
+  }
+
+  while (lowest_block <= highest_block) {
+    let search_block = Math.floor((lowest_block + highest_block) / 2);
+    contract_code = await provider.getCode(contract_address, search_block);
+    if (contract_code != '0x') {
+      highest_block = search_block;
+    } else if (contract_code == '0x') {
+      lowest_block = search_block;
+    }
+
+    if (highest_block == lowest_block + 1) {
+      return highest_block;
+    }
+  }
```

Comment:
Hi @chimpydev There is limit on number of requests a strategy can send https://github.com/snapshot-labs/snapshot-strategies#checklist-for-adding-a-new-strategy
Please find alternate ways to get this data

Reply:
Yes sir but as you have on other strategy, they are only used with one argument on examples.json
This is only to be used for each address itself as argument
@[acrylicfiddle]

Reply:
No i mean this can send many requests to find the correct contract creator, and can load up score API server, better to find someother good way to get this in a single request

Reply:
Try this https://ethereum.stackexchange.com/a/12904/91696